src/components/CompactFarm/CompactFarm.tsx

```diff
@@ -122,7 +122,7 @@ export class CompactFarm extends CommonComponent<ICompactFarmProps, any> {
                                 status={server.status}
                                 roleEdit={this._onRoleChange}
                                 showEditRoles={this.props.showEditRoles}
-                                onClose={server.onClose}
+                                onClose={this.props.serverClose !== undefined ? this._onServerClose : undefined}
```

Comment:
personally i like the **on** prefix in the prop names when dealing with events. I can tell immediately that it is some event handler